### PR TITLE
Initial Ray runner prototype based on FnApiRunner in Beam

### PR DESCRIPTION
**Pull request by pabloem**
This prototype is hacked together as quickly as possible. I can't speak for its efficiency, and its code cleanliness.

Next steps:
- Enable more and more tests under `ray_runner_test.py`
- Make efficiency improvements 
    - Creation of pcollection holding actors (probably rearchitect)
    - Data transfer and encoding
    - Improve context passing (serialization and passing-around of RayRunnerExecutionContext, RayBundleContext)

Others...

### CODE REVIEW

**Comment by pabloem** [comment]
fyi @pdames @jjyao et al

File: ray_beam_runner/ray_runner_test.py [hunk 1]
```diff
@@ -84,6 +85,18 @@ def contains_labels(mi, labels):
     return contains_labels(mi, labels) and mi.urn == urn
 
 
+class RayFnRunnerTest(fn_runner_test.FnApiRunnerTest):
```
Comment (jjyao) [inline ray_beam_runner/ray_runner_test.py:88]: Is this used?

**Review by pdames — COMMENTED** [review]
Awesome work - thanks Pablo! 

It may be a good idea to start capturing some of the TODOs and comments here as issues for others to start picking up.

I was able to get the Ray Runner tests passing with one change related to `WatermarkManager` initialization. Can you verify whether or not this is expected? I just had a couple other minor comments, but otherwise think this looks ready to merge!

**Comment by wilsonwang371** [comment]
LGTM. Pablo, can we have a demo on this later sometime?

**Review by pdames — APPROVED** [review]
I've committed the minor change required to make supported tests pass - merging. Thanks, Pablo!

**Comment by pabloem** [comment]
thanks Patrick, everyone. I'll try to create new work items for this.

@wilsonwang371 I'll be happy to do a demo on this.

**Pull request merged**